src/Azure/AzureAD/Authentication.AzureAD.UI/src/AzureADJwtBearerOptionsConfiguration.cs

```diff
@@ -30,8 +30,10 @@ public void Configure(string name, JwtBearerOptions options)
                 return;
             }
 
-            options.Audience = azureADOptions.ClientId;
-            options.Authority = new Uri(new Uri(azureADOptions.Instance), azureADOptions.TenantId).ToString();
+            options.Audience = string.Format(azureADOptions.Audience?.Replace("{ClientId}", "{0}"),
```

Comment:
This has several problems.
* What happens if azureAdOptions.Audience is null? (It's not ok to show explicitly on the template config file).
* We don't do replacements this way. `azureADOptions.Audience?.Replace("{ClientId}", "{0}"`
  * Normally we use a string interpolation.
* Same for authority below.

Reply:
@javiercn : how can we use string interpollation for a string which is not in the code, but in a config file? could you please show me an example on how to do it?

Reply:
@jmprieur how is azureADOptions populated?

Reply:
@javiercn is there a way to enforce some type of validation, For example, azureADOptions.Audience must be populated correctly from config?